src/PrivateComposerInstaller/Plugin.php

```diff
@@ -60,82 +160,153 @@ public function activate(Composer $composer, IOInterface $io)
     }
 
     /**
-     * Injects package version into dist url if it contains placeholders.
-     * @param PackageEvent $event
+     * @inheritdoc
+     */
+    public function deactivate(Composer $composer, IOInterface $io)
+    {
+        $this->composer = null;
+        $this->io = null;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function uninstall(Composer $composer, IOInterface $io)
+    {
+        //
+    }
+
+    /**
+     * Handle PRE_PACKAGE_INSTALL and PRE_PACKAGE_UPDATE Composer events.
+     *
+     * Only gets triggered running Composer 1.
+     *
+     * @param \Composer\Installer\PackageEvent $event
+     *
+     * @return void
+     */
+    public function handlePreInstallUpdateEvent(PackageEvent $event): void
+    {
+        $operation = $event->getOperation();
+        $package = $operation->getJobType() === 'update'
+            ? $operation->getTargetPackage()
+            : $operation->getPackage();
+
+        // If running Composer 1 the package version needs to be injected
+        // before the dist URL gets written into `composer.lock`
+        $distUrl = $package->getDistUrl();
+        $version = $package->getPrettyVersion();
+        $filteredDistUrl = $this->fulfillVersionPlaceholder($distUrl, $version);
+        if ($filteredDistUrl !== $distUrl) {
+            $package->setDistUrl($filteredDistUrl);
+        }
+    }
+
+    /**
+     * Fulfill package URL placeholders before downloading the package.
+     *
+     * @param \Composer\Plugin\PreFileDownloadEvent $event
+     *
      * @return void
      */
-    public function injectVersion(PackageEvent $event): void
+    public function handlePreDownloadEvent(PreFileDownloadEvent $event): void
     {
-        $package = $this->getOperationPackage($event->getOperation());
-        $url = $package->getDistUrl();
+        $filteredProcessedUrl = $processedUrl = $event->getProcessedUrl();
```

Comment:
```suggestion
        $filteredProcessedUrl = $filteredCacheKey = $processedUrl = $event->getProcessedUrl();
```
Extra variable to store the version-fulfilled URL.